test/TestRegistry.spec.ts

```diff
@@ -0,0 +1,390 @@
+
+      const avatar = await registry.avatarOf(a.user1);
+      expect(a.user1).to.be.equal(await registry.ownerOf(avatar));
+      expect(avatar).to.be.equal(await registry.avatarOf(a.user1));
+    });
+
+    it("should create different avatar for each users", async () => {
+      const avatar1 = await registry.avatarOf(a.user1);
+      expect(avatar1).to.be.not.equal(ZERO_ADDRESS);
+      expect(a.user1).to.be.equal(await registry.ownerOf(avatar1));
+      expect(avatar1).to.be.equal(await registry.avatarOf(a.user1));
+
+      const tx = await registry.newAvatar({ from: a.user2 });
+      const avatar2 = await registry.avatarOf(a.user2);
+      expectEvent(tx, "NewAvatar", { avatar: avatar2, owner: a.user2 });
+      expect(avatar2).to.be.not.equal(ZERO_ADDRESS);
+      expect(a.user2).to.be.equal(await registry.ownerOf(avatar2));
+      expect(avatar2).to.be.equal(await registry.avatarOf(a.user2));
+
+      expect(avatar1).to.be.not.equal(avatar2);
+    });
+  });
+
+  describe("getAvatar()", async () => {
+    it("should create new if not exists for a user", async () => {
+      let avatar3 = await registry.avatarOf(a.user3);
+      expect(ZERO_ADDRESS).to.be.equal(avatar3);
+
+      avatar3 = await registry.getAvatar.call(a.user3, { from: a.user3 });
+      const tx = await registry.getAvatar(a.user3, { from: a.user3 });
+      expectEvent(tx, "NewAvatar", { avatar: avatar3, owner: a.user3 });
+      expect(avatar3).to.be.not.equal(ZERO_ADDRESS);
+    });
+
+    it("should allow anyone to create avatar for others", async () => {
+      let avatar4 = await registry.avatarOf(a.user4);
+      expect(ZERO_ADDRESS).to.be.equal(avatar4);
+
+      avatar4 = await registry.getAvatar.call(a.user4, { from: a.other });
+      const tx = await registry.getAvatar(a.user4, { from: a.other });
+      expectEvent(tx, "NewAvatar", { avatar: avatar4, owner: a.user4 });
+      expect(avatar4).to.be.not.equal(ZERO_ADDRESS);
+    });
+
+    it("should return avatar if exists for a user", async () => {
+      const avatar1 = await registry.getAvatar(a.user1);
+      expect(avatar1).to.be.not.equal(ZERO_ADDRESS);
+
+      const avatar2 = await registry.getAvatar(a.user2);
+      expect(avatar2).to.be.not.equal(ZERO_ADDRESS);
+    });
+
+    it("should allow anyone to get avatar of others", async () => {
+      const avatar1 = await registry.getAvatar(a.user1, { from: a.other });
+      expect(avatar1).to.be.not.equal(ZERO_ADDRESS);
+    });
+
+    it("should fail when zero address is passed", async () => {
+      let avatar = await registry.avatarOf(ZERO_ADDRESS);
+      expect(avatar).to.be.equal(ZERO_ADDRESS);
+
+      await expectRevert(
+        registry.getAvatar(ZERO_ADDRESS, { from: a.other }),
+        "Registry: owner-address-is-zero",
+      );
+
+      avatar = await registry.avatarOf(ZERO_ADDRESS);
+      expect(avatar).to.be.equal(ZERO_ADDRESS);
+    });
+  });
+
+  describe("delegateAvatar()", async () => {
```

Comment:
at later point we need to add also `functional` test here, where delgatee is doing stuff on behalf of the user.
Or maybe it would be in the avatar test. but somewhere.

Reply:
yes it will part of Avatar testing.